src/qt/breakpoint_window.hpp

```diff
@@ -0,0 +1,81 @@
+#ifndef BREAKPOINT_WINDOW_H
+#define BREAKPOINT_WINDOW_H
+
+#include <QDialog>
+#include <QtCore/QVariant>
+#include <QtWidgets/QApplication>
+#include <QtWidgets/QCheckBox>
+#include <QtWidgets/QComboBox>
+#include <QtWidgets/QDialog>
+#include <QtWidgets/QDialogButtonBox>
+#include <QtWidgets/QGridLayout>
+#include <QtWidgets/QLabel>
+#include <QtWidgets/QLineEdit>
+#include <QtWidgets/QSpacerItem>
+#include "../core/int128.hpp"
+#include "../core/ee/emotion_breakpoint.hpp"
+
+namespace Ui {
+class breakpoint_window;
+}
+
+class breakpoint_window : public QDialog
+{
+    Q_OBJECT
+
+public:
+    explicit breakpoint_window(QWidget *parent = 0);
```

Comment:
To be explicit, `nullptr` should be used here, unless there is a specific reason 0 is being used?

```suggestion
    explicit breakpoint_window(QWidget *parent = nullptr);
```